src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -88,6 +98,43 @@ private static void ResolveExecutablePath(ref string executable, ref string args
                 args = $"/C \"{executable}\"";
                 executable = comSpec;
             }
+
+            return executable;
+        }
+
+        private static string ResolveExecutablePathFromProject(string executable, NuGetFramework framework)
+        {
+            if (framework == null) return null;
+
+            var projectRootPath = PathUtility.GetProjectRootPath();
+
+            if (projectRootPath == null) return null;
+
+            var commandName = Path.GetFileNameWithoutExtension(executable);
+
+            var projectContext = ProjectContext.Create(projectRootPath, framework);
+
+            var commandPackage = projectContext.LibraryManager.GetLibraries()
```

Comment:
What's the convention? Any file in the package that has deps and an executable/dll?